### DISCUSSION TITLE
Issue with certain keywords in x-for loop

### DISCUSSION THREAD

**Discussion by wayz9** [alpinejs/alpine#1977] [7. Bugs (archived)] [answered]
Today I'v encountered a weird behaviour with `x-for` . In my example I have json array of cases (like support tickets) and I wish to display them using alpine and x-for loop. 

So I declared a `div`  with `x-data="getCases()"`  and at the end of my document I declared that function that returns cases.

```
<script> 
 function getCases() {
  return {
    cases: [
       {
         id: 59102,
         title: 'Cool Title',
         // and other things
       },
        // more items
     ]
   }
}
</script>
```

Next thing was of course to itterate through this via `template` and `x-for`.

```
<template x-for="case in cases">
  ...
</template>
```

But this is when it actually breaks, because of naming, so if I put `case in cases` it thinks that I used `case` instead of word case.

![Screenshot 2021-08-22 233237](https://user-images.githubusercontent.com/71568418/130370646-dfc2054b-d28d-422f-a542-ba2d4a03298e.png)

Tried it with other keywords like `class in classes`, `function in functions` also produces same error and breaks. 

So to fix thing I just have to name things differently and it should work, anyway just wanted to share this with the community. 

Don't know if this should be threated like a big thing.

**Comment by SimoTod** [comment, marked as answer]
It's probably worth highlighting it for people who don't know javascript but yeah, Alpine strings are just pieces of javascript and like any other languages there are reserved words you can't use as variable name because they have a specific meaning. It would be like calling a variable `if`.
If it helps, here's the list of reserved words: https://www.w3schools.com/js/js_reserved.asp
If it helps
